client/my-sites/plans/trials/trial-acknowledge/acknowlege-modal.tsx

```diff
@@ -1,18 +1,75 @@
+import { PLAN_HOSTING_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { Site } from '@automattic/data-stores';
+import { NEW_HOSTED_SITE_FLOW, createSiteWithCart } from '@automattic/onboarding';
 import { Modal } from '@wordpress/components';
+import useAddHostingTrialMutation from 'calypso/data/hosting/use-add-hosting-trial-mutation';
+import { useSelector } from 'calypso/state';
+import { getCurrentUserName } from 'calypso/state/current-user/selectors';
 import { HostingTrialAcknowledgement } from './hosting-acknowledge';
 
 type TrialAcknowledgeModalProps = {
 	setOpenModal: ( open: boolean ) => void;
 };
 
 export const TrialAcknowledgeModal = ( { setOpenModal }: TrialAcknowledgeModalProps ) => {
+	const { mutateAsync: addHostingTrial } = useAddHostingTrialMutation();
+	const username = useSelector( ( state ) => getCurrentUserName( state ) );
+
+	const createTrialSite = async () => {
+		const sourceSlug = 'source-slug-to-be-added';
+		const selectedSiteTitle = 'selected-site-title-to-be-added';
+		const isPaidDomainItem = false;
+		const useThemeHeadstart = false;
+		const domainItem = null;
+		const domainCartItem = null;
+		const theme = 'pub/zoologist';
+
+		const site = await createSiteWithCart(
```

Comment:
I don't think we need to create a site here, since the site is already existing. we only need to `await addHostingTrial( { siteId: site.siteId, planSlug: PLAN_HOSTING_TRIAL_MONTHLY } );` which will add the trial plan in the background and make it atomic. We then need to poll the atomic status determine when it is finished

Reply:
I'll pick it up tomorrow if you didn't get to it

Reply:
Yeah, that was my first thought on it. Updated yesterday still. Thanks!